jexl-eval/src/lib.rs

```diff
@@ -312,100 +433,204 @@ mod tests {
+            }),
+        );
+        assert_eq!(evaluator.eval("'T_T'|lower").unwrap(), value!("t_t"));
+    }
+
+    #[test]
+    // Test returning an UnknownTransform error if a transform is unknown
+    fn test_missing_transform() {
+        let err = Evaluator::new().eval("'hello'|world").unwrap_err();
+        if let EvaluationError::UnknownTransform(transform) = err {
+            assert_eq!(transform, "world")
+        } else {
+            panic!("Should have thrown an unknown transform error")
+        }
+    }
+
+    #[test]
+    fn test_add_multiple_transforms() {
+        let mut transforms = HashMap::new();
+        transforms.insert(
+            "square".to_string(),
+            TransformFn::from(Box::new(|v: &[Value]| {
+                let num = v
+                    .first()
+                    .expect("There should be one argument!")
+                    .as_f64()
+                    .expect("Should be a valid number!");
+                Ok(value!((num as u64).pow(2)))
+            })),
+        );
+        transforms.insert(
+            "sqrt".to_string(),
+            TransformFn::from(Box::new(|v: &[Value]| {
+                let num = v
+                    .first()
+                    .expect("There should be one argument!")
+                    .as_f64()
+                    .expect("Should be a valid number!");
+                Ok(value!(num.sqrt() as u64))
+            })),
+        );
+        let evaluator = Evaluator::new().with_transforms(transforms);
+        assert_eq!(evaluator.eval("4|square").unwrap(), value!(16));
+        assert_eq!(evaluator.eval("4|sqrt").unwrap(), value!(2));
+        assert_eq!(evaluator.eval("4|square|sqrt").unwrap(), value!(4));
+    }
+
+    #[test]
+    fn test_transform_with_argument() {
+        let evaluator = Evaluator::new().with_transform(
+            "split",
+            Box::new(|args: &[Value]| {
+                let s = args
+                    .first()
+                    .expect("Should be a first argument!")
+                    .as_str()
+                    .expect("Should be a string!");
+                let c = args
+                    .get(1)
+                    .expect("There should be a second argument!")
+                    .as_str()
+                    .expect("Should be a string");
+                let res: Vec<&str> = s.split_terminator(c).collect();
+                Ok(value!(res))
+            }),
+        );
+
+        assert_eq!(
+            evaluator.eval("'John Doe'|split(' ')").unwrap(),
+            value!(vec!["John", "Doe"])
+        );
+    }
+
+    #[derive(Debug, thiserror::Error)]
+    enum CustomError {
+        #[error("Invalid argument in transform!")]
+        InvalidArgument,
+    }
+
+    #[test]
+    fn test_custom_error_message() {
+        let evaluator = Evaluator::new().with_transform(
+            "error",
+            Box::new(|_: &[Value]| Err(CustomError::InvalidArgument.into())),
```

Comment:
A hack to allow consumers to use their own custom errors.

It goes like this:
`CustomError` -> `anyhow::Error` -> `EvaluatorError::CustomError`